service/collector/cloudformation.go

```diff
@@ -0,0 +1,166 @@
+	clientaws "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// Label key that will hold the CF name.
+	labelCloudFormation = "cloudformation"
+
+	// Second part of the metric name, right after namespace.
+	subsystemCloudFormation = "cloudformation"
+)
+
+var (
+	cloudFormationStackDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystemCloudFormation, "info"),
+		"CloudFormation information.",
+		[]string{
+			labelAccountID,
+			labelAccountType,
+			labelCluster,
+			labelID,
+			labelInstallation,
+			labelName,
+			labelOrganization,
+			labelStack,
+			labelState,
+		},
+		nil,
+	)
+)
+
+// Configuration struct.
+type CloudFormationConfig struct {
+	Helper *helper
+	Logger micrologger.Logger
+
+	InstallationName string
+}
+
+// Main struct for this collector.
+type CloudFormation struct {
+	helper *helper
+	logger micrologger.Logger
+
+	installationName string
+}
+
+// Creates a new CloudFormation metrics collector.
+func NewCloudFormation(config CloudFormationConfig) (*CloudFormation, error) {
+	if config.Helper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Helper must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.InstallationName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.InstallationName must not be empty", config)
+	}
+
+	cf := &CloudFormation{
+		helper: config.Helper,
+		logger: config.Logger,
+
+		installationName: config.InstallationName,
+	}
+
+	return cf, nil
+}
+
+// Collect is the main metrics collection function.
+func (cf *CloudFormation) Collect(ch chan<- prometheus.Metric) error {
+	awsClientsList, err := cf.helper.GetAWSClients()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var g errgroup.Group
+
+	for _, item := range awsClientsList {
+		awsClients := item
+
+		g.Go(func() error {
+			err := cf.collectForAccount(ch, awsClients)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		})
+	}
+
+	err = g.Wait()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// Describe emits the description for the metrics collected here.
+func (cf *CloudFormation) Describe(ch chan<- *prometheus.Desc) error {
+	ch <- cloudFormationStackDesc
+	return nil
+}
+
+// collectForAccount collects and emits metrics for one AWS account.
+func (cf *CloudFormation) collectForAccount(ch chan<- prometheus.Metric, awsClients clientaws.Clients) error {
+	o, err := awsClients.CloudFormation.DescribeStacks(&cloudformation.DescribeStacksInput{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	accountID, err := cf.helper.AWSAccountID(awsClients)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	accountType := "tenant-cluster"
```

Comment:
Why are we certain here, that it is `tenant-cluster`? And if we are certain, why do we need this label?

We can keep it, I'd just like to understand as we don't have it for e.g. VPC metrics.

Reply:
How I understood it, awsClients only contains the tenant cluster accounts and I added the label to make sure that later the control plane accounts can be included, too and we can seperate them.
However, I just double checked and it seems like both accounts are actually already included. I'll make sure they are labeled correctly!